### PR TITLE
Add source Olaoe "Ar"

### PR DESCRIPTION
**Pull request by mohamedotaku**
Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [ ] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: multisrc/overrides/madara/olaoe/src/Olaoe.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    data class SearchResponseDto(
+        val data: List<SearchEntryDto>,
+        val success: Boolean,
+    )
+
+    @Serializable
+    data class SearchEntryDto(
+        val url: String = "",
+        val title: String = "",
+    )
```
Comment (beer-psi) [inline multisrc/overrides/madara/olaoe/src/Olaoe.kt]: Why did you need to override this? Madara search seems to work fine.

File: multisrc/overrides/mangathemesia/thunderscans/res/web_hi_res_512.png
Comment (beer-psi) [inline multisrc/overrides/mangathemesia/thunderscans/res/web_hi_res_512.png:1]: > Have removed `web_hi_res_512.png` when adding a new extension

File: multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    override val seriesTypeSelector = "span:contains(النوع) a"
+    override val seriesStatusSelector = "span:contains(الحالة)"
+
+    override fun pageListParse(document: Document): List<Page> {
+        val scriptContent = document.selectFirst("script:containsData(ts_reader)")!!.data()
+        val jsonString = scriptContent.substringAfter("ts_reader.run(").substringBefore(");")
+        val tsReader = json.decodeFromString<TSReader>(jsonString)
+        val imageUrls = tsReader.sources.firstOrNull()?.images ?: return emptyList()
+        return imageUrls.mapIndexed { index, imageUrl -> Page(index, document.location(), imageUrl) }
+    }
```
Comment (beer-psi) [inline multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt:71]: The `MangaThemesia` base class can already handle pages from `ts_reader`.

File: multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    override fun chapterListSelector() = "div.bxcl li, ul div:has(span.lchx)"
+
+    override fun chapterFromElement(element: Element) = SChapter.create().apply {
+        val urlElements = element.select("a")
+        setUrlWithoutDomain(urlElements.attr("href"))
+        name = element.select(".lch a, .chapternum").text().ifBlank { urlElements.last()!!.text() }
+        date_upload = element.selectFirst(".chapter-date")?.text().parseChapterDate()
```
Comment (beer-psi) [inline multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt:79]: Isn't the selector `.chapterdate` on the website, which would make this override unneeded?

File: multisrc/overrides/madara/olaoe/res/web_hi_res_512.png
Comment (beer-psi) [inline multisrc/overrides/madara/olaoe/res/web_hi_res_512.png:1]: > Have removed `web_hi_res_512.png` when adding a new extension

**Review by beer-psi — CHANGES_REQUESTED** [review]

File: multisrc/overrides/madara/olaoe/src/Olaoe.kt (hunk 1 above)
Comment (mohamedotaku) [reply multisrc/overrides/madara/olaoe/src/Olaoe.kt]: ok thats work fine thanks

File: multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt [hunk 4]
```diff
@@ -0,0 +1,120 @@
+package eu.kanade.tachiyomi.extension.ar.thunderscans
```
Comment (stevenyomi) [inline multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt:1]: Put this source in another PR. Only one source per PR.

File: multisrc/overrides/madara/olaoe/src/Olaoe.kt [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+            setOnPreferenceChangeListener { _, _ ->
+                Toast.makeText(screen.context, RESTART_TACHIYOMI, Toast.LENGTH_LONG).show()
+                true
+            }
+        }
+        screen.addPreference(baseUrlPref)
+
+        super.setupPreferenceScreen(screen)
+    }
```
Comment (stevenyomi) [inline multisrc/overrides/madara/olaoe/src/Olaoe.kt]: Don't add base URL setting, unless the site changes URL really often.

File: multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                Toast.makeText(screen.context, RESTART_TACHIYOMI, Toast.LENGTH_LONG).show()
+                true
+            }
+        }
+        screen.addPreference(baseUrlPref)
+
+        super.setupPreferenceScreen(screen)
+    }
+
+    private fun getPrefBaseUrl(): String = preferences.getString(BASE_URL_PREF, defaultBaseUrl)!!
```
Comment (stevenyomi) [inline multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt:119]: Same for this.

File: multisrc/overrides/madara/olaoe/src/Olaoe.kt (hunk 5 above)
Comment (mohamedotaku) [reply multisrc/overrides/madara/olaoe/src/Olaoe.kt]: ok i will fix it

File: multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt (hunk 4 above)
Comment (mohamedotaku) [reply multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt:1]: ok i will remove it for now

File: multisrc/overrides/madara/olaoe/src/Olaoe.kt (hunk 5 above)
Comment (mohamedotaku) [reply multisrc/overrides/madara/olaoe/src/Olaoe.kt]: Ok i will fix it

File: multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt (hunk 6 above)
Comment (mohamedotaku) [reply multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt:119]: ok i will fix it in next PR

File: multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt (hunk 3 above)
Comment (mohamedotaku) [reply multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt:79]: ok thanks

File: multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt (hunk 2 above)
Comment (mohamedotaku) [reply multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt:71]: ah ok thanks

File: multisrc/overrides/madara/olaoe/res/web_hi_res_512.png
Comment (mohamedotaku) [reply multisrc/overrides/madara/olaoe/res/web_hi_res_512.png:1]: ok
